### DISCUSSION TITLE
Question about `kitty.conf` doc (resetting the terminal)

### DISCUSSION THREAD

**Discussion by xfzv** [kovidgoyal/kitty#6684] [Q&A] [answered]
I'm using the following function in my `zsh` config to both scroll the current screen contents into the scrollback buffer and clear the screen, as suggested on [Kitty website](https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Reset-the-terminal):

```zsh
ctrl_l() {
    builtin print -rn -- $'\r\e[0J\e[H\e[22J' >"$TTY"
    builtin zle .reset-prompt
    builtin zle -R
}
zle -N ctrl_l
bindkey '^l' ctrl_l
```

It works fine. However, when I'm chrooting to my second Linux distro on my machine, the function's very first command gives me an error:

```zsh
$ builtin print -rn -- $'\r\e[0J\e[H\e[22J' >"$TTY"                                                                               
zsh: permission denied: /dev/pts/1
```

Removing `>"$TTY"` seems to fix the problem, the function works as expected in chroot without it:

```zsh
builtin print -rn -- $'\r\e[0J\e[H\e[22J'                              
```

Could someone explain why the command is redirected to `$TTY` in the doc example? Is there any drawback not doing so?

Thanks.

**Comment by kovidgoyal** [comment, marked as answer]
It's done so that it works even if stdout is not a tty. If you want to
replace itm better to use /dev/tty rather than removing it completely.

On Sat, Oct 07, 2023 at 01:11:17PM -0700, xfzv wrote:
> I'm using the following function in my `zsh` config to both scroll the current screen contents into the scrollback buffer and clear the screen, as suggested on [Kitty website](https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Reset-the-terminal):
> 
> ```zsh
> ctrl_l() {
>     builtin print -rn -- $'\r\e[0J\e[H\e[22J' >"$TTY"
>     builtin zle .reset-prompt
>     builtin zle -R
> }
> zle -N ctrl_l
> bindkey '^l' ctrl_l
> ```
> 
> It works fine. However, when I'm chrooting to my second Linux distro on my machine, the function's very first command gives me an error:
> 
> ```zsh
> $ builtin print -rn -- $'\r\e[0J\e[H\e[22J' >"$TTY"                                                                               
> zsh: permission denied: /dev/pts/1
> ```
> 
> Removing `>"$TTY"` seems to fix the problem, the function works as expected in chroot without it:
> 
> ```zsh
> builtin print -rn -- $'\r\e[0J\e[H\e[22J'                              
> ```
> 
> Could someone explain why the command is redirected to `$TTY` in the doc example? Is there any drawback not doing so?
> 
> Thanks.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6684
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________
